lib/widgets/sample_viewer_page.dart

```diff
@@ -132,22 +132,41 @@ class _SampleViewerPageState extends State<SampleViewerPage> {
   }
 
   void onSearchChanged(String searchText) {
-    final List<Sample> results;
+    var results = <Sample>[];
+    // restore the initial list of samples if the search text is empty
     if (searchText.isEmpty) {
-      results = _allSamples;
+      if (widget.category == null) {
+        results = [];
+      } else if (widget.category == SampleCategory.all) {
+        results = _allSamples;
+      } else if (widget.category != null) {
+        results = getSamplesByCategory(widget.category);
+      }
     } else {
-      results = _allSamples.where(
-        (sample) {
-          final lowerSearchText = searchText.toLowerCase();
-          return sample.title.toLowerCase().contains(lowerSearchText) ||
-              sample.category.toLowerCase().contains(lowerSearchText) ||
-              sample.keywords.any(
-                (keyword) => keyword.toLowerCase().contains(lowerSearchText),
-              );
-        },
-      ).toList();
+      if (widget.category == null || widget.category == SampleCategory.all) {
+        results = _allSamples.where(
+          (sample) {
+            final lowerSearchText = searchText.toLowerCase();
+            return sample.title.toLowerCase().contains(lowerSearchText) ||
+                sample.category.toLowerCase().contains(lowerSearchText) ||
+                sample.keywords.any(
+                  (keyword) => keyword.toLowerCase().contains(lowerSearchText),
+                );
+          },
+        ).toList();
+      // if the category is not null, the only samples within the category are searched
+      } else if (widget.category != null) {
```

Comment:
Likewise the `if (widget.category != null)` is redundant here, because the first branch of the `if` statement is taking care of the case where `widget.category == null`.